### DISCUSSION TITLE
Metro Release TRV11 2.x Discussion

### DISCUSSION THREAD

**Discussion by 92shreyansh** [ONDC-Official/mobility-specification#63] [General]
Hi All,

Lets use this thread to provide feedback and comments on the draft 2.x version of TRV11 use cases

**Comment by abha-sinha** [comment]
Hello

I have a few queries related to the Metro APIs:

1.Select

 

a.What does item id refer to? Is it same as category id?
 

2.On_select

 

a.Is the provider time same as that sent for on_search call or different?
b.What is the significance of time under items
c.Relation among item id, item category id & fulfillement id; can an item have different categories.
d.fulfillment ids for item is same as fulfillment id or different
 

3.init

 

a.billing details is the details of the customer?
b.Billing is a mandatory attribute, but email, nae & phone are optional?
c.What is the payment status & payment type?
 

4.Confirm

 

a. Can the buyer send confirm API before receiving On_init?

 

 5.On_confirm

 

a.  Status- Possible values?

b. How does the fulfillment information changes from On_Search API, to On_init to On_Confirm

c.  Does the Quote change with On_Search,On_init& On_confirm API?

d. How does updated_at time differ fro that of created_at?

 

6.On_Status

 Does this refer to the status of ticket or that of metro?

**Reply by amitsinha07** [reply]
Answers:
1. Select
     a. item id is a unique id for each item and it is not the same as the category id. Category id is a unique id for each 
     category.

2. On_select
    a. Yes it is the same as the on_search call Because we select it from the on_search call so it will be the same for the entire flow

    b. In items time gives us information about the validity of that item. For example, validity is PT2D. It means it is valid for 2 days.

    c. Items: describes a product or a service offered to the end consumer by the provider. In the mobility sector, it can 
        represent a fare of a product like a one-way journey. In the logistics sector, it can represent the delivery service 
        offering. In the retail domain, it can represent a product like a grocery item.
        Fulfillment: describes how an order will be rendered/fulfilled to the end-customer
        Category: describes a category of a vehicle
        Yes, items can have multiple category ids and multiple fulfilment ids.

    d. Yes, fulfillment ids for items are the same as fulfillment id.

3. init
    a. Yes, billing details are details of a customer. 

    b. Yes billing is mandatory but email, name and phone are optional

    c. Payment status denotes whether payment is made or not (PAID or NOT-PAID) and 
        payment type is when the customer has to pay. In payment type have three options in case of mobility 
            1. PRE-ORDER: Payment occurs before order placement.
            2. ON-FULFILLMENT: Payment occurs upon completion of the journey.
            3. POST-FULFILLMENT: Payment occurs after the completion of the journey.

4. confirm
    a. No, you can't confirm the order before initialising the order

5. on_confirm 
    a. order status describes the status of the order. Possible values are
          1. ACTIVE 
          2. COMPLETE 
          3. SOFT_CANCEL
          4. CONFIRM_CANCEL
          5. CANCELLED
          
     b. In on_search fulfillment type is ROUTE and in on_select it changes to TRIP. In on_confirm authorization and tags with 
         TICKET_INFO object   is added to payload

     c. Quote is not present in on_search call. The quote is attached to the payload after the on_init call and it is the same in 
         on_confirm.

     d. created_at is the time of order creation and updated_at is the time when the order was last updated 

7. On_Status
    a. status refers to an order status

**Comment by sharique-zafar** [comment]
updated_at will always be greater than created_at

**Comment by Janak2401** [comment]
What is the base url for metro ?
